lib/main_common.pm

```diff
@@ -197,6 +197,10 @@ sub setup_env {
         set_var('ORIGINAL_TARGET_VERSION', get_var('VERSION'));
         set_var('VERSION',                 get_var('UPGRADE_TARGET_VERSION'));
     }
+    # Set correct VERSION for the base system setup on z/VM
+    if (get_var('BASE_VERSION') && !is_upgrade() && check_var('ARCH', 's390x')) {
```

Comment:
Alright, as we are talking about zVM here, we need this initial value there only we should use `check_var('BACKEND', 's390x')` instead. For me it's still confusing to have `!is_upgrade` as this is used for migration tests. Nevertheless, `BACKEND` variable is not used for fresh installations, so this change, should not break anything.
Have you also considered just setting this variable in the test suite.
If you have migrated job group to yaml, you can set machine specific settings per test suite, e.g.:
```
 - offline_sles15_pscc_basesys-srv_all_full_zVM_preparation:
        machine: s390x-zVM-hsi-l3
        settings:
          VERSION: %BASE_VERSION%
```
This should do the same thing you do in the code here.

Reply:
Hi Rodion, I will close this PR since I found there is an existing method for me to solve this issue. Just set UPGRADE_TARGET_RELEASED_VERSION=1 UPGRADE_TARGET_VERSION=15 will set the version to SLE15 even this case in the SLE15SP2 job group.
Anyway, for your confuse, I will try to explain, that is, the first step of the zVM test is to install SLE15 so this shouldn't set UPGRADE=1.
Very thanks for the comments and show me the example to set setting in YAML.